src/python/bot/fuzzers/libFuzzer/engine.py

```diff
@@ -372,23 +370,62 @@ def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
-
-    merge_result = runner.merge(
-        [output_dir] + input_dirs,
-        merge_timeout=max_time,
-        tmp_dir=merge_tmp_dir,
-        additional_args=arguments,
-        artifact_prefix=reproducers_dir)
-
-    if merge_result.timed_out:
-      raise MergeError('Merging new testcases timed out')
-
-    if merge_result.return_code != 0:
-      raise MergeError('Merging new testcases failed')
+    merge_control_file = os.path.join(merge_tmp_dir, 'MCF')
+
+    # Make sure the merge control file does not exist at this point.
+    shell.remove_file(merge_control_file)
+
+    additional_args = arguments + [
+        '%s%s' % (constants.MERGE_CONTROL_FILE_ARGUMENT, merge_control_file)
+    ]
+    merge_stats = {}
+
+    # Two step merge process to obtain accurate stats for the new corpus units.
+    # See https://reviews.llvm.org/D66107 for a more detailed description.
+    def _merge_step(corpus_directories, step):
+      """Execute the fuzz target, check the result, record the stats."""
+      result = runner.merge(
+          corpus_directories,
+          merge_timeout=max_time,
+          tmp_dir=merge_tmp_dir,
+          additional_args=additional_args,
+          artifact_prefix=reproducers_dir)
+
+      if result.timed_out:
+        raise MergeError('Merging new testcases (step %d) timed out' % step)
+
+      if result.return_code != 0:
+        raise MergeError('Merging new testcases (step %d) failed' % step)
+
+      step_stats = stats.parse_stats_from_merge_log(result.output.splitlines())
+      return result, step_stats
+
+    # Step 1. Use only existing corpus and collect "initial" stats.
+    corpus_directories = [output_corpus_dir] + existing_corpus_dirs
+    result_1, stats_1 = _merge_step(corpus_directories, 1)
+    merge_stats['initial_edge_coverage'] = stats_1['edge_coverage']
+    merge_stats['initial_feature_coverage'] = stats_1['feature_coverage']
+
+    # Clear the output dir as it does not have any new units at this point.
+    engine_common.recreate_directory(output_corpus_dir)
+
+    # Step 2. Process the new corpus units as well.
+    corpus_directories.append(new_corpus_dir)
+    result_2, stats_2 = _merge_step(corpus_directories, 2)
+    merge_stats['edge_coverage'] = stats_2['edge_coverage']
+    merge_stats['feature_coverage'] = stats_2['feature_coverage']
+
+    # Diff the stats to obtain accurate values for the new corpus units.
+    merge_stats['new_edges'] = (
+        merge_stats['edge_coverage'] - merge_stats['initial_edge_coverage'])
+    merge_stats['new_features'] = (
+        merge_stats['feature_coverage'] -
+        merge_stats['initial_feature_coverage'])
 
     # TODO(ochang): Get crashes found during merge.
-    return engine.FuzzResult(merge_result.output, merge_result.command, [], {},
-                             merge_result.time_executed)
+    return engine.FuzzResult(result_1.output + result_2.output,
+                             result_2.command, [], merge_stats,
```

Comment:
result_2.command feels a bit wrong here since we really are doing two. But I don't have anything better to suggest.

Reply:
yeah... the only difference is the last (new) corpus dir, so it seems fine to log the final command only